server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -264,8 +264,14 @@ public void addLong(String fieldName, long value) {
         @Override
         public void addUnsignedLong(String fieldName, long value) {
             try (BytesStreamOutput out = new BytesStreamOutput()) {
-                out.write((byte) 'u');
-                out.writeLong(value);
+                Object ul = DocValueFormat.UNSIGNED_LONG_SHIFTED.format(value);
+                if (ul instanceof Long l) {
+                    out.write((byte) 'l');
+                    out.writeLong(l);
+                } else {
+                    out.write((byte) 'u');
+                    out.writeLong(value);
+                }
```

Comment:
@nik9000 I change the code, if the value is Long, use 'l' to encode it. If the value is too long, the value is a BigInteger type, so use the 'u' to encode it.
So, when decode the value, small value will decode to be Long, long value will decode to be BigInteger.